backend/api/Controllers/MissionController.cs

```diff
@@ -384,25 +385,44 @@ public async Task<ActionResult<byte[]>> GetMap([FromRoute] string installationCo
         if (scheduledMissionQuery.AreaName != null)
             area = await _areaService.ReadByInstallationAndName(scheduledMissionQuery.InstallationCode, scheduledMissionQuery.AreaName);
 
-        var scheduledMissionDefinition = new MissionDefinition
+        var source = await _sourceService.ReadBySourceIdAndType($"{echoMission.Id}", MissionSourceType.Echo);
+        MissionDefinition? missionDefinition = null;
+
+        if (source == null)
         {
-            Id = Guid.NewGuid().ToString(),
-            Source = new Source
+            source = new Source
             {
                 SourceId = $"{echoMission.Id}",
                 Type = MissionSourceType.Echo
-            },
-            Name = echoMission.Name,
-            InspectionFrequency = scheduledMissionQuery.InspectionFrequency,
-            InstallationCode = scheduledMissionQuery.InstallationCode,
-            Area = area
-        };
+            };
+        }
+        else
+        {
+            var missionDefinitions = await _missionDefinitionService.ReadMissionDefinitionsBySourceId(source.SourceId);
+            // In theory there could be more than one mission definition that is relevant, but we need to randomly choose one
```

Comment:
Following the discussion on having a one-to-one mapping between missionDefinition and Source, this can be refactored